packages/cli/lib/utils.js

```diff
@@ -103,8 +103,13 @@ exports.validateClassName = function(name) {
 
 /**
  * Validate project directory to not exist
+ * and check if the directory path is based on CWD
  */
 exports.validateNotExisting = function(projDir) {
+  const invalidPath = projDir.split('/').includes('~');
```

Comment:
I think we can disable `~` as a whole, for example, not allowing `~xyz`.

Reply:
I was not sure about that since `~` could be part of the folder name. But yeah if the folder name contains `~`, `lb4 app` would complain about it. What do you think?

Reply:
The value `~` is a special path interpreted by unix shells as the home directory, which is defined in the environment variable `$HOME`. Instead of rejecting `~` in the path, I'd prefer to interpret it the same way as the shell does:
 - On Unix systems like Linux and MacOS, replace it with `process.env.HOME`.
- On Windows, the value should be replaced with `process.env.userprofile`, see https://stackoverflow.com/questions/9228950/what-is-the-alternative-for-users-home-directory-on-windows-command-prompt

As for Raymond's proposal to disallow `~xyz`, it seems like an arbitrary limitation with no good reason behind it and as such I'd prefer to avoid such limitation.

Reply:
I don't think it's our CLI's responsibility to support shell conventions for a few reasons:

1. Windows users might interpret such chars differently, for example, not considering `~` as home directory

2. If we decide to support `~` for Unix, what about other shell features, such as env vars? IMO, `~` is just a shortcut, our users can always use the full path if necessary.

3. I think Unix shell only allows `~` or `~/<other>`.

Reply:
I agree that it's not our CLI's responsibility to support shell conventions. But I will go with whichever @rmg and @bajtos think is more reasonable.
We can warn the user about the usage of `~`. 
Or we can replace it with `process.env.HOME || process.env.userprofile`.

Reply:
Not sure if I was mentioned by typo or not, but I have opinions so I'll share them.

From a UX perspective, I personally would expect to be able to use `~` paths here because I do it in other interactive terminal programs all the time (eg. `vim`).

Proper expansion of `~` is surprisingly limited in scope and defined for only a few cases:

- leading `~`, bare or followed by `/`, expands to current user's home directory
  - eg. `~`, `~/`, `~/foo`
- leading `~` followed by a username expands to the given user's home directory
  - eg. `~rmg`, `~rmg/`, `~rmg/bar`

A `~` appearing anywhere other than as the first character (eg. `/~/foo`, `foo/~/bar`) should be treated as a literal `~` and not expanded or interpreted.

The simple case (current user only) can actually be completely handled inline:

```js
str.replace(/^~(?=\/|$)/, os.homedir())
```

Expanding the `~username` form would require some very OS-specific calls that make it a lot more complicated. But I'm not sure if anyone would ever use that form with this CLI, so I suspect nobody would even notice if it wasn't supported.

(A quick search reveals [untildify](https://github.com/sindresorhus/untildify), which confirms both the simplicity of the easy case and the complexity of the difficult case.)

Simplicity of adding it directly to the CLI aside, I think this sort of thing would actually make the most sense as part of inquirer itself.

Reply:
@raymondfeng

Reply:
I'm fine to use https://github.com/sindresorhus/untildify to handle `~`.

Reply:
+1 to use untildify